src/EnsembleKalmanInversions.jl

```diff
@@ -258,9 +266,28 @@ function iterate!(eki::EnsembleKalmanInversion;
 end
 
 #####
-##### Stepping and adaptive stepping
+##### Failure condition, stepping, and adaptive stepping
 #####
 
+struct NormExceedsMedian{T}
+    minimum_relative_norm :: T
+end
+
+NormExceedsMedian() = NormExceedsMedian(1e9)
```

Comment:
```suggestion
"""
    struct NormExceedsMedian{T}

The particle failure condition type. A particle is marked "failed" if the forward map norm is
larger than `minimum_relative_norm` times more than the median value for the ensemble.
Default value `minimum_relative_norm = 1e9`.
"""
struct NormExceedsMedian{T}
    minimum_relative_norm :: T
    NormExceedsMedian(x) = x < 0 ? error("minimum_relative_norm must non-negative") : new{typeof(x)}(x)
    NormExceedsMedian() = new{Float64}(1e9)
end
```

Reply:
feel free to ignore this suggestion... or take bits of it...

Reply:
It's great! But I think we should document the constructors and not the types, because users don't care about the type definition. What do you think?

Reply:
Also why use inner constructors? I think we should always use outer constructors if necessary. Sometimes it's nice to have a performant, non-input-validating inner constructor available for low-level stuff too. This case may not matter much.

Reply:
have a look at [5d02545](https://github.com/CliMA/ParameterEstimocean.jl/pull/273/commits/5d02545c18f99db61ca60abd60b9de8ab87cd7a5)
Merge state
Add more commits by pushing to the [glw/failure-criteria](https://github.com/CliMA/ParameterEstimocean.jl/tree/glw/failure-criteria) branch on [CliMA/ParameterEstimocean.jl](https://github.com/CliMA/ParameterEstimocean.jl).

This branch has not been deployed
No deployments
Show all reviewers
Changes approved
1 approving review by reviewers with write access. [Learn more.](https://docs.github.com/articles/about-pull-request-reviews/)
1 approval
 Hide all checks
Some checks haven’t completed yet
6 queued checks
@github-actions
CI / Julia 1.6 - ubuntu-latest - x64 - pull_request (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/CliMA/ParameterEstimocean.jl/runs/6747457999?check_suite_focus=true)
@github-actions
Documentation / build (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/CliMA/ParameterEstimocean.jl/runs/6747457998?check_suite_focus=true)
@github-actions
CI / Julia 1.6 - ubuntu-latest - x86 - pull_request (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/CliMA/ParameterEstimocean.jl/runs/6747458045?check_suite_focus=true)
@github-actions
CI / Julia 1.6 - macOS-latest - x64 - pull_request (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/CliMA/ParameterEstimocean.jl/runs/6747458075?check_suite_focus=true)
@github-actions
CI / Julia 1.6 - windows-latest - x64 - pull_request (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/CliMA/ParameterEstimocean.jl/runs/6747458112?check_suite_focus=true)
@github-actions
CI / Julia 1.6 - windows-latest - x86 - pull_request (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/CliMA/ParameterEstimocean.jl/runs/6747458141?check_suite_focus=true)
This branch has no conflicts with the base branch
Merging can be performed automatically.
Merge pull request  You can also [open this in GitHub Desktop](https://desktop.github.com/) or view  command line instructions .
@navidcy

Write   Preview
 


Attach files by dragging & dropping, selecting or pasting them.
  Close pull request
Comment
Remember, contributions to this repository should follow our [GitHub Community Guidelines](https://docs.github.com/articles/github-community-guidelines).
 ProTip! Add [.patch](https://github.com/CliMA/ParameterEstimocean.jl/pull/273.patch) or [.diff](https://github.com/CliMA/ParameterEstimocean.jl/pull/273.diff) to the end of URLs for Git’s plaintext views.
Reviewers
 @[navidcy](https://github.com/navidcy)
navidcy
 

Still in progress? Convert to draft
Assignees
No one—assign yourself
Labels
None yet
Projects
None yet
Milestone
No milestone
Development
Successfully merging this pull request may close these issues.

https://github.com/CliMA/ParameterEstimocean.jl/issues/272
Notifications
Customize
 Unsubscribe
You’re receiving notifications because your review was requested.
2 participants
@glwagner
@navidcy
 Lock conversation